standards/ICRC-2/README.md

```diff
@@ -46,16 +48,17 @@ type Account = record {
 
 ### icrc2_approve
 
-Entitles the `spender` to transfer an additional token `amount` on behalf of the caller from account `{ owner = caller; subaccount = from_subaccount }`.
+This method entitles the `spender` to transfer token `amount` on behalf of the caller from account `{ owner = caller; subaccount = from_subaccount }`.
 The number of transfers the `spender` can initiate from the caller's account is unlimited as long as the total amounts and fees of these transfers do not exceed the allowance.
 The caller does not need to have the full token `amount` on the specified account for the approval to succeed, just enough tokens to pay the approval fee.
-The `spender`'s allowance for the account increases or decreases by the `amount` depending on the sign of the `amount` field. 
-If the `expires_at` field is not null, the ledger resets the approval expiration time to the specified value.
+The call resets the allowance and the expiration date for the `spender` account to the given values.
+
+The ledger SHOULD reject the call if the spender account is equal to the source account (`{ owner = caller; subaccount = from_subaccount } == spender`).
```

Comment:
```suggestion
The ledger SHOULD reject the call if the spender account owner is equal to the source account owner.
```

Reply:
Initially, I restricted the rejection to only the same account because it might simplify the implementation of some ledgers (the ICP ledger, for example).

In a ledger that works with AccountIdentifiers, it's hard to check whether an account belongs to a particular principal. 

It's still possible to implement the check you propose; we have to do it much earlier before the conversion to AccountIdentifier happens. I'm OK either way.

Reply:
I see how it can simplify the implementation of ledgers that work with account-identifiers if the implementation first converts icrc accounts to account-identifiers before the main method logic however since the icrc2 standard works with visible account owners and the icrc2_approve method request contains the approver/caller account owner-principal and the spender account owner-principal, I rather the standard make the check. If a caller of icrc2_approve can approve to him/her self, why not let the approver-account be the same as the spender account?